tests/phpunit/tests/db.php

```diff
@@ -2472,4 +2472,91 @@ public function data_parse_db_host() {
 			),
 		);
 	}
+
+	/**
+	 * @ticket 55414
+	 */
+	public function test_suppress_query_error() {
```

Comment:
There appear to be four tests being performed here:

1. Testing that the value of suppress errors is sent to the filter (false)
2. Testing that the value of suppress errors is sent to the filter (true)
3. Testing that error suppression respects the value returned by filter callbacks (false).
4. Testing that error suppression respects the value returned by filter callbacks (true).

These should be two separate tests, each with a data provider.

1. Testing that the value of suppress errors is sent to the filter (false/true)
```php
/**
 * Tests that the current value of `suppress_errors` is sent to the 'suppress_query_error' filter.
 *
 * @ticket 55414
 *
 * @covers wpdb::print_error
 *
 * @dataProvider data_suppress_errors
 *
 * @param bool $suppress_errors Whether to suppress errors.
 */
public function test_wpdb_print_error_should_pass_suppress_errors_to_suppress_query_error_filter( $suppress_errors ) {
}
```

2. Testing that error suppression respects the value returned by filter callbacks (false/true).
```php
/**
 * Tests that `wpdb::print_error()` respects the value returned from the 'suppress_query_error' filter.
 *
 * @ticket 55414
 *
 * @covers wpdb::print_error
 *
 * @dataProvider data_suppress_errors
 *
 * @param bool $suppress_errors Whether to suppress errors.
 */
public function test_wpdb_print_error_should_respect_suppress_query_error_filtered_value( $suppress_errors ) {
}
```

Data provider:
```php
/**
 * Data provider.
 *
 * @return array[]
 */
public function data_suppress_errors() {
	return array(
		'false' => array( 'suppress_errors' => false ),
		'true'  => array( 'suppress_errors' => true ),
	);
}
```

Reply:
Thank you for this. The tests are much cleaner and clearer when separated and using data providers for different configurations.

For the second test, I wanted to be more complete in the permutations tested, so I used a different data provider for that test.